dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/BacklashHysteresis.mo

```diff
@@ -0,0 +1,69 @@
+within Dynawo.NonElectrical.Blocks.NonLinear;
+
+/*
+* Copyright (c) 2022, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite
+* of simulation tools for power systems.
+*/
+
+block BacklashHysteresis "Provide a region of offset with a hysteresis logical"
+  import Modelica;
+
+  extends Modelica.Blocks.Interfaces.SISO;
+
+  parameter Boolean H0 = false "Value of h at initial time";
+  parameter Real U0 "Start value of input";
+  parameter Real UHigh "Upper limit of the hysteresis";
+  parameter Real ULow = -UHigh "Lower limit of the hysteresis";
+
+  final parameter Real Aux0 = if H0 then max(U0 - UHigh, 0) else min(U0 - ULow, 0) "Initial auxiliary variable";
+
+  Real aux(start = Aux0) "Auxiliary variable to stabilize y when the derivative of u changes";
+  Boolean h(start = H0) "Boolean to detect changes of behaviour of u";
+
+equation
+  when der(u) < -1e-5 and u >= UHigh then
```

Comment:
Not a big fan of this 1e-5. We could maybe add a parameter Tolerance

Reply:
I agree